### ISSUE TITLE
2.2.3 changes console.log to print

### ISSUE THREAD

**Issue by MMhunter** [NeekSandhu/onigasm#19]
![image](https://user-images.githubusercontent.com/1378130/68732735-71b22d80-060f-11ea-9449-a8539db0a8b2.png)

this snippet changes console.log to print, causing problems when we attempt to log something.

**Comment by NeekSandhu** [comment]
Can you please point where exactly this code lives?

cc @aeschli

**Comment by MMhunter** [comment]
@NeekSandhu  It is in the `lib/onigasm.js`  after build.  Version 2.2.2 does not have this.

**Comment by aeschli** [comment]
Looks like an emscripten bug, introduced by https://github.com/emscripten-core/emscripten/commit/c45d3b545529ae5c6abfe93d7639a5aeb4caabc1

**Comment by aeschli** [comment]
This happens only when `print` is defined:
```
if (typeof print !== "undefined") { 
        if (typeof console === "undefined") 
            console = {}; 
        console.log = print; 
        console.warn = console.error = typeof printErr !== "undefined" ? printErr : print 
      }
```
@MMhunter Is that the case in your runtime environment?

**Comment by aeschli** [comment]
I filed an issue against emscripten: https://github.com/emscripten-core/emscripten/issues/9829
and pushed a workaround on our side (restore console.log/error/warn). Also added a test.

Published onigasm@2.2.4 with the fix. Please give it a try!
